pkg/controllers/clusteroperator_controller.go

```diff
@@ -97,20 +97,8 @@ func (r *CloudOperatorReconciler) Reconcile(ctx context.Context, _ ctrl.Request)
 		return ctrl.Result{}, err
 	}
 
-	platform, err := config.GetProviderFromInfrastructure(infra)
```

Comment:
Why this check was removed?

Reply:
Agreed, this check should say so that compose config can take the platform directly still, otherwise you could hit nil panics there